File: MRMLasso_matlab/MRMLasso/computeLpri2.m
function [values pri_terms2] = computeLpri2(Beta,U,J,lambdaR,lambdaS,train_label,train_data)

s=length(train_data);
m=length(train_label);

pri_terms2 = [];

% term1: (1/2m)||Y-\sum_v diag(F^v*U{v}')||_2^2
tmp = zeros(m,1);
for v = 1:s
    vec = diag(train_data{v}*U{v}');
    tmp = tmp + vec;
end
term = norm(train_label - tmp);
term = 0.5/m*term*term;
pri_terms2 = [pri_terms2 term];

% term2
term = 0;
n = 0;
for v = 1:s
    term = term + norm(Beta{v},1);
    n = n + length(Beta{v});
end
%term = term * lambdaS/n;
term = term * lambdaS/s;
pri_terms2 = [pri_terms2 term];

% term3
sval = svd(J);
term = lambdaR*sum(sval);
pri_terms2 = [pri_terms2 term];

values = sum(pri_terms2);

File: MRMLasso/computeLpri2.py
import numpy as np

def computeLpri2(m , s , Beta , U , J ,lambdaR , lambdaS , train_label , train_data):

    pri_term2 = []

    # term1
    tmp = np.zeros((m , 1))
    for v in range(s):
        vec = np.diag(train_data[v] * U[v].T)
        tmp = tmp + vec
    term = np.linalg.norm(train_label - tmp , ord=2)
    term = 0.5/m*term*term
    pri_term2.append(term)

    # term2
    term = 0
    n = 0
    for v in range(s):
        term = term + np.linalg.norm(Beta[v] , ord=1)
        n = n + len(Beta[v])
    term = term * lambdaS / s
    pri_term2.append(term)

    # term3
    sval = np.linalg.svd(J , compute_uv=False)
    term = lambdaR * sum(sval)
    pri_term2.append(term)

    values = sum(pri_term2)

    return values , pri_term2